test/TesterInternal/StorageTests/PersistenceGrainTests.cs

```diff
@@ -1200,45 +1177,58 @@ private void SetErrorInjection(string providerName, ErrorInjectionPoint errorInj
                 //}
             }
         }
-
-        private MockStorageProvider FindStorageProviderInUse(string providerName, bool okNull = false)
+        private bool CanGetStorageProviderInUse(string providerName)
         {
-            MockStorageProvider providerInUse = null;
-            SiloHandle siloInUse = null;
-            List<SiloHandle> silos = this.HostedCluster.GetActiveSilos().ToList();
-            foreach (var siloHandle in silos)
+            foreach (var siloHandle in this.HostedCluster.GetActiveSilos())
             {
-                MockStorageProvider provider = (MockStorageProvider)siloHandle.AppDomainTestHook.GetStorageProvider(providerName);
-                Assert.NotNull(provider);
-                if (provider.ReadCount > 0)
+                if (siloHandle.TestHook.CanGetStorageProvider(providerName).Result)
                 {
-                    Assert.Null(providerInUse);
-                   
-                    providerInUse = provider;
-                    siloInUse = siloHandle;
```

Comment:
yes tests are passing.

The old testing approach this test suit is taking:
1. Get reference of the StorageProviderInUse 
2. manipulate its state using its reference directly
3. test different cases regarding its changed state

Since TestHook cannot return the reference of StorageProviderInUse anymore. So the new approach we are taking
1. determine if the provider exists, probably not very needed anymore in some test cases, just as you mentioned
2. using IControllable to change storage providers state indirectly by name, in all silos
3. test different cases regarding changing.